crates/steel-parser/src/ast.rs

```diff
@@ -977,6 +997,60 @@ impl From<Require> for ExprKind {
     }
 }
 
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+pub struct Vector {
+    pub args: Vec<ExprKind>,
+    pub bytes: bool,
+    pub span: Span,
+}
```

Comment:
A single struct for both vectors and bytevectors. It's not too cumbersome and saves us from adding a `visit_bytevector()` method in a zillion places.